src/Raven.Server/Extensions/JsonSerializationExtensions.cs

```diff
@@ -1,48 +1,69 @@
-﻿using Raven.Client.Documents.Indexes;
+﻿using Microsoft.AspNetCore.Http;
+using Raven.Client.Documents.Indexes;
 using Raven.Client.Documents.Indexes.Spatial;
 using Sparrow.Json.Parsing;
 
 namespace Raven.Server.Extensions
 {
     public static class JsonSerializationExtensions
     {
+        public static DynamicJsonValue ToJson(this IQueryCollection queryCollection)
+        {
+            var jsonMap = new DynamicJsonValue();
+            if (queryCollection == null) //precaution, prevent NRE
+                return null;
+
+            foreach (var kvp in queryCollection)
+            {
+                jsonMap[kvp.Key] = kvp.Value.ToArray();
```

Comment:
You are always returning an array here, but in most cases, this will actually only have a single value for property